File: rust/src/leetcode/twoSum.rs
use std::collections::HashMap;

pub fn two_sum(nums: Vec<i32>, target: i32) -> Option<Vec<i32>> {
    let mut map: HashMap<i32, usize> = HashMap::new();

    for (i, v) in nums.iter().enumerate() {
        let diff = target - v;

        if map.contains_key(&diff) {
            let k = map.get(&diff).unwrap();
            return Some(vec![*k as i32, i as i32]);
        }
        map.insert(*v, i);
    }
    None
}

File: Python/algoritms/leetCode/twoSum.py
def twoSum(nums: list[int], target: int) -> list[int] | None:
    prevMap = {}

    for i, n in enumerate(nums):
        diff = target - n
        if diff in prevMap:
            return [prevMap[diff], i]
        prevMap[n] = i


print(twoSum([3, 2, 4], 6))
